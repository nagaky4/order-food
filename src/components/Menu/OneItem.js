import React from 'react'
import { Link } from 'react-router-dom';

import ComposeAddToCart from '../HOC/addToCart'


const OneItem = (props) => {
    return (

        <div className="col-3" key={props.value.id}>
            <div className="card text-center mt-2 mb-2">
                <Link to={'/menu/' + props.value.id} >
                    <img className="card-img-top " src={props.value.images} alt="" />
                </Link>
                <div className="card-body">
                    <h4 className="card-title">{props.value.name}</h4>
                    <p>{props.value.price} VNƒê</p>

                    {props.children}

                </div>
            </div>
        </div >
    )
}




// var mapDispatchToProps = (dispatch, props) => {
//     return {
//         addToList: (item) => {
//             dispatch(actions._addToList(item));
//         }
//     }
// }


export default ComposeAddToCart(OneItem);
