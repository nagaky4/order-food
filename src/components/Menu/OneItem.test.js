import React from 'react';
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { OneItem } from './OneItem';

configure({ adapter: new Adapter() });


const setUp = (props = {}) => {
    return shallow(
        <Router>
            <OneItem {...props} />
        </Router>
    )
}

describe('<One Item />', () => {
    let wrapper;
    describe('Have props', () => {

        beforeEach(() => {
            const props = {
                value: {
                    id: "1",
                    name: 'Bánh xèo',
                    price: '50000',
                    images: 'https://beptueu.vn/hinhanh/tintuc/top-15-hinh-anh-mon-an-ngon-viet-nam-khien-ban-khong-the-roi-mat-2.jpg',
                    detail: 'Thuộc top 40 món ngon nhất Việt Nam do CNN bình chọn. Bánh xèo là món ngon bạn khó cưỡng lại được.'
                },
                convertVND: () => {

                }
            }
            wrapper = setUp(props)
        });


        it('should be render without error', () => {
            expect(wrapper.length).toBe(1);
        });

        it('snapshot renders', () => {
            const component = renderer.create(wrapper);
            let tree = component.toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('should recive props is object', () => {

            expect(wrapper.find('OneItem').props().value).toEqual({
                id: "1",
                name: 'Bánh xèo',
                price: '50000',
                images: 'https://beptueu.vn/hinhanh/tintuc/top-15-hinh-anh-mon-an-ngon-viet-nam-khien-ban-khong-the-roi-mat-2.jpg',
                detail: 'Thuộc top 40 món ngon nhất Việt Nam do CNN bình chọn. Bánh xèo là món ngon bạn khó cưỡng lại được.'
            })
        })

    })
    describe('Not props', () => {

        beforeEach(() => {
            wrapper = setUp();
        });

        it('should not render', () => {
            expect(wrapper.find('div.col-4')).toHaveLength(0)
        });
    })

})