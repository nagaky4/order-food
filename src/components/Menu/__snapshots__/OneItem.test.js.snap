// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<One Item /> Have props snapshot renders 1`] = `
<div
  className="col-4"
>
  <div
    className="card text-center mt-2 mb-2 one-item"
  >
    <a
      href="/menu/1"
      onClick={[Function]}
    >
      <img
        alt=""
        className="undefined card-img-top"
        src="https://beptueu.vn/hinhanh/tintuc/top-15-hinh-anh-mon-an-ngon-viet-nam-khien-ban-khong-the-roi-mat-2.jpg"
      />
    </a>
    <div
      className="card-body"
    >
      <h4
        className="undefined card-title"
      >
        Bánh xèo
      </h4>
      <p>
         VNĐ
      </p>
    </div>
  </div>
</div>
`;
